### ISSUE TITLE
Points fill and cursor marker issues in dark mode

### ISSUE THREAD

**Issue by tve** [leeoniya/uPlot#780]
With a white background the std points fill and cursor markers work fine. However, on a dark background they don't.
Repro: https://codepen.io/codesurfr/pen/vYaNwwR
- this is a copy of https://leeoniya.github.io/uPlot/demos/grid-over-series.html with the body background changed to #1e1e1e
- the points size is enlarged to 10pts so one can see stuff better
- the red series has no other change, notice how the points are filled with white instead of the background color
- the blue and green series fix that by setting the points fill to the same as the background color, however now the cursor marker no longer works properly because it uses the points fill color
- it is possible to completely fix the issue by defining the cursor points fill explicitly, see commented out code in the codepen

Is this a desired/intended state of the built-in cursor? Dark backgrounds are pretty popular for plots...

Maybe there should be an `opts.dark` or `opts.background_color` type of option passed to uPlot so it can do something more sensible by default?

Quick screen grab showing the points (no cursor here):
![image](https://user-images.githubusercontent.com/39480/209580566-628ae43a-4594-4077-9a92-6c39b1bdde6d.png)

**Comment by leeoniya** [comment]
yeah this was a tough call to make when it was implemented. ideally i wanted to be able to punch a donut hole through the existing line but i dont think this can be done cheaply (maybe with some globalCompositeOperation trickery, though seems unlikely).

> it is possible to completely fix the issue by defining the cursor points fill explicitly, see commented out code in the codepen

i'll think about what to do here. but this relatively low priority since the fix is straightforward.

also...

```js
        cursor: {
          points: {
            fill: (u, sidx) => { return u.series[sidx].fill()},
            size: (u, sidx) => u.series[sidx].points.size,
          },
        },
```

this works around the hover point sizing issue (i'm at 1.5 `devicePixelRatio` these days, so may have overlooked it on previous setup with 1.0 ratio):

[1] https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation

**Comment by leeoniya** [comment]
https://github.com/leeoniya/uPlot/commit/c8d541ea25bab3c6c111ac51267cbda90351342c should fix the hover point sizing issue.
